internal/service/kafka/resource_kafka_topic.go

```diff
@@ -330,13 +331,27 @@ func getKafkaTopicConfig(d *schema.ResourceData) aiven.KafkaTopicConfig {
 	}
 }
 
-func resourceKafkaTopicRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+func resourceKafkaTopicReadOrCreate(ctx context.Context, d *schema.ResourceData, m interface{}, isResource bool) diag.Diagnostics {
 	project, serviceName, topicName, err := schemautil.SplitResourceID3(d.Id())
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
-	topic, err := getTopic(ctx, d, m, false)
+	topic, err := getTopic(ctx, d, m)
+
+	// Topics are destroyed when kafka is off
+	// https://docs.aiven.io/docs/platform/concepts/service-power-cycle
+	// So it's better to recreate them, than make user to clear the state manually
+	// Recreates missing topics:
+	// 1. if server returns 404
+	// 2. only for "resource" type (not datasource)
+	// 3. only for resources with existing state, not imports
+	if errors.Is(err, errTopicNotFound) && isResource && !d.IsNewResource() {
```

Comment:
Btw, we already have this logic down below in the next if statement:
```
if err != nil {
  return diag.FromErr(schemautil.ResourceReadHandleNotFound(err, d))
}
```

So I think we can drop this IF and moreover we can reuse it in data-source, no need for a separate functions for each case. Data-source read function will not error I think in case of `d.SetId("")`

Reply:
Nice. It didn't work because cache overrides error, I'll try to fix it.